modules/directory/init.zsh

```diff
@@ -28,3 +28,19 @@ unsetopt CLOBBER            # Do not overwrite existing files with > and >>.
 
 alias d='dirs -v'
 for index ({1..9}) alias "$index"="cd +${index}"; unset index
+
+if (is-at-least 4.1) \
+    && zstyle -t ':prezto:module:directory' persist-dirstack
+then
+    if [[ -f ~/.zdirs ]] && [[ ${#dirstack[*]} -eq 0 ]]; then
+        echo "\n" >> ~/.zdirs
+        dirstack=( ${(uf)"$(< ~/.zdirs)"//"$HOME"[[:space:]]/} )
```

Comment:
Ok, `(uf)`:

- f: Split the result of the expansion at newlines.
- u: Expand only the first occurrence of each unique word.

Then `//` appears to be find and replace all matches... Is this to make sure the home directory isn't in the dirstack? Should we really be removing the home dir if the user added it via pushd (or another method)?

Maybe I'm dense, but I'm still not sure how this line accomplishes the main goal of loading the list from `~/.zdirs` or why all these random flags are needed. Could anyone clarify?